rpcs3/Emu/VFS.cpp

```diff
@@ -13,16 +13,17 @@
 #endif
 
 #include <thread>
+#include <unordered_map>
 
 LOG_CHANNEL(vfs_log, "VFS");
 
 struct vfs_directory
 {
 	// Real path (empty if root or not exists)
-	std::string path{};
+	std::string path;
 
-	// Virtual subdirectories (vector because only vector allows incomplete types)
-	std::vector<std::pair<std::string, vfs_directory>> dirs{};
+	// Virtual subdirectories
+	std::unordered_map<std::string, std::unique_ptr<vfs_directory>> dirs;
```

Comment:
if you are already using uniqure ptr, then revert back to vector. savestates utilities here rely on the constant ordering of elements which unordered_map dies not provide.

Reply:
or just use std::map